src/pew/pygobject_gtk/__init__.py

```diff
@@ -0,0 +1,73 @@
+import logging
+import marshal
+
+import gi
+gi.require_version('Gio', '2.0')
+gi.require_version('Gtk', '3.0')
+from gi.repository import GLib, Gio, Gtk
+
+
+def run_on_main_thread(func, *args, **kwargs):
+    def source_fn(*args, **kwargs):
+        func(*args, **kwargs)
+        return False
+
+    data = (args, kwargs)
+    main_context = GLib.MainContext.default()
+    main_context.invoke_full(GLib.PRIORITY_DEFAULT, source_fn, *args, **kwargs)
+    return None
+
+
+app = None
+
+
+def choose_file(callback):
+    print("choose_file", callback)
```

Comment:
Could we use `logging.warning` statements instead, so that when running as a GUI these would get logged?

Reply:
Oh, sure! And I just realized I forgot to implement that, didn't I?

Reply:
It's okay with me to leave some things unimplemented, especially if you have no use case yourself or no easy way to test. As long as the logs warn about it, I think that's good. I'd also be okay with raising `NotImplementedError`.